src/tools/tracejob.c

```diff
@@ -382,21 +382,45 @@ void
 parse_log(FILE *fp, char *job, int ind)
 {
 	struct log_entry tmp;	/* temporary log entry */
-	char buf[16384];	/* buffer to read in from file */
+	char *buf;		/* buffer to read in from file */
+	char *tbuf;		/* help wth larger entries in log */
 	char job_buf[128];	/* hold the jobid and the . */
 	char *p;		/* pointer to use for strtok */
 	int field_count;	/* which field in log entry */
 	int j = 0;
-	struct tm tms;	/* used to convert date to unix date */
+	struct tm tms;		/* used to convert date to unix date */
 	int lineno		= 0;
 	int slen;
 	char *pdot;
+	int buf_size = 16384;	/* initial buffer size */
+	int tbuf_size = buf_size;
+	int break_fl = 0;
+
+	buf = (char*)malloc(buf_size * sizeof(char));
+	tbuf = (char*)malloc(buf_size * sizeof(char));
 
 	tms.tm_isdst = -1;	/* mktime() will attempt to figure it out */
 
 	strcpy(job_buf, job);
 
-	while (fgets(buf, 16384, fp) != NULL) {
+	while (fgets(buf, buf_size, fp) != NULL) {
+		while(buf_size == strlen(buf) + 1) {
+			buf_size *= 2;
+			buf = (char*)realloc(buf, (buf_size + 1) * sizeof(char));
+			if(buf == NULL || fgets(tbuf, tbuf_size + 1, fp) == NULL){
+				break_fl = 1;
+				break;
+			}
+			strcat(buf, tbuf);
+			tbuf_size = buf_size;
+			tbuf = (char*)realloc(tbuf, (tbuf_size + 1) * sizeof(char));
+			if(tbuf == NULL){
+				break_fl = 1;
+				break;
+			}
+		}
+		if(break_fl)
+			break;
 		lineno++;
 		j++;
 		buf[strlen(buf)-1] = '\0';
```

Comment:
You're absolutely right. fgets() does handle null termination. I noticed this line but decided not to touch it since it does no apparent harm. Since you mentioned, I'll remove it.